pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go

```diff
@@ -179,42 +179,47 @@ var _ = Describe("Validating VMICreate Admitter", func() {
 	)
 
 	Context("tolerations with eviction policies given", func() {
-		var vmi *v1.VirtualMachineInstance
-		var policyMigrate = v1.EvictionStrategyLiveMigrate
-		var policyMigrateIfPossible = v1.EvictionStrategyLiveMigrateIfPossible
-		var policyNone = v1.EvictionStrategyNone
-		var policyExternal = v1.EvictionStrategyExternal
-
 		BeforeEach(func() {
 			enableFeatureGate(deprecation.LiveMigrationGate)
-			vmi = api.NewMinimalVMI("testvmi")
-			vmi.Spec.EvictionStrategy = nil
 		})
 
-		DescribeTable("it should allow", func(policy *v1.EvictionStrategy) {
-			vmi.Spec.EvictionStrategy = policy
-			resp := ValidateVirtualMachineInstanceSpec(k8sfield.NewPath("fake"), &vmi.Spec, config)
-			Expect(resp).To(BeEmpty())
+		DescribeTable("it should allow", func(policy v1.EvictionStrategy) {
+			vmi := newBaseVmi(libvmi.WithEvictionStrategy(policy))
+
+			ar, err := newAdmissionReviewForVMICreation(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			resp := vmiCreateAdmitter.Admit(context.Background(), ar)
+
+			Expect(resp.Allowed).To(BeTrue())
 		},
-			Entry("migration policy to be set to LiveMigrate", &policyMigrate),
-			Entry("migration policy to be set None", &policyNone),
-			Entry("migration policy to be set External", &policyExternal),
-			Entry("migration policy to be set to LiveMigrateIfPossible", &policyMigrateIfPossible),
-			Entry("migration policy to be set nil", nil),
+			Entry("migration policy to be set to LiveMigrate", v1.EvictionStrategyLiveMigrate),
+			Entry("migration policy to be set None", v1.EvictionStrategyNone),
+			Entry("migration policy to be set External", v1.EvictionStrategyExternal),
+			Entry("migration policy to be set to LiveMigrateIfPossible", v1.EvictionStrategyLiveMigrateIfPossible),
 		)
 
 		It("should allow no eviction policy to be set", func() {
```

Comment:
How does this case differ from `Entry("migration policy to be set nil", nil),` that was removed?

Reply:
It doesn't, the reason it was moved to a separate `It` is because `libvmi.WithEvictionStrategy` takes by value meaning the Describe Table parameter can't be nil.

Reply:
Solid compromise, although I find that it slightly detracts from readability. I'll proceed with it anyhow as it does consolidate the cases nicely.

Reply:
I've just noticed an older term is used here (`migration policy`) and it should be replaced with `eviction strategy`.

Reply:
Done.